### PR TITLE
feature/authentication-api

### PR DESCRIPTION
**Pull request by lui297**
Changes:
- added an authentication action and mutation to store;
- added an authenticationresponsemodel;
- fixed the login connection on the view page;
- added an api builder for future environment variables.
- lint error hotfix

### CODE REVIEW

File: vue.config.js [hunk 1]
```diff
@@ -16,4 +16,9 @@ module.exports = {
   configureWebpack: {
     devtool: "source-map"
   },
+  devServer: {
+    host: '127.0.0.1',
```
Comment (jeanpaulkommers) [inline vue.config.js:20]: Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/Supershift/Sushi.Mediakiwi.UI/pullRequest?prid=8560396)
```suggestion
    host: "127.0.0.1",
```

File: vetur.config.js [hunk 2]
```diff
@@ -1,7 +1,9 @@
 // vetur.config.js
 /** @type {import('vls').VeturConfig} */
 module.exports = {
+  
   projects: [
+    './src',
```
Comment (jeanpaulkommers) [inline vetur.config.js:6]: Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/Supershift/Sushi.Mediakiwi.UI/pullRequest?prid=8560396)
```suggestion
    "./src",
```

File: vue.config.js (hunk 1 above)
Comment (jeanpaulkommers) [inline vue.config.js:20]: Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/Supershift/Sushi.Mediakiwi.UI/pullRequest?prid=8560396)
```suggestion
    host: "127.0.0.1",
```

File: src/components/ResetPassword.vue [hunk 3]
```diff
@@ -68,15 +72,23 @@ export default defineComponent({
       value: contentResetPassword.value.resetButtonText,
       readOnly: false,
     });
-    function handleSubmit() {
-      if (customEmailInput.value.fieldValue && validEmail.value) {
-        store.dispatch("resetPassword", customEmailInput.value.fieldValue);
+    function handleReset() {
+      /* eslint no-console:0 */
+      console.log("email.value, validEmail.value");
```
Comment (jeanpaulkommers) [inline src/components/ResetPassword.vue:77]: No console log

File: src/components/ResetPassword.vue [hunk 4]
```diff
@@ -68,15 +72,23 @@ export default defineComponent({
...
+      
+      if (email.value && validEmail.value) {
+        /* eslint no-console:0 */
+        console.log(customEmailInput.value.fieldValue);
```
Comment (jeanpaulkommers) [inline src/components/ResetPassword.vue:81]: no console log

**Review by jeanpaulkommers — CHANGES_REQUESTED** [review]
Fix all linter errors
Maybe move the API calls like _getMediakiwiContentAPI_, _getTopNavigationAPI_, etc to the api.ts
the store is getting **very** big

File: src/store/modules/Content/mutations.ts [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+
+export type Mutations<S = ContentState> = {
+  [MutationTypes.SET_CONTENT](state: S, payload: GetContentMediakiwiResponseModel): void,
+  [MutationTypes.SET_GRIDS](state: S, payload: Grid[]): void,
+  [MutationTypes.SET_FOLDERS](state: S, payload: Field[]): void,
+  [MutationTypes.SET_BUTTONS](state: S, payload: ButtonModel[]): void,
+  [MutationTypes.SET_PAGE](state: S, payload: PageModel): void,
+  [MutationTypes.SET_FIELDS](state: S, payload: Field[]): void,
+  [MutationTypes.SET_RESOURCES](state: S, payload: ResourceModel[]): void,
+  [MutationTypes.SET_POST_CONTENT](state: S, payload: PostContentMediakiwiResponseModel|any): void,
```
Comment (jeanpaulkommers) [inline src/store/modules/Content/mutations.ts:18]: Codacy has a fix for the issue: [Unexpected any. Specify a different type.](https://app.codacy.com/gh/Supershift/Sushi.Mediakiwi.UI/pullRequest?prid=8560396)
```suggestion
  [MutationTypes.SET_POST_CONTENT](state: S, payload: PostContentMediakiwiResponseModel|unknown): void,
```

File: src/components/form/FormComponent.vue [hunk 6]
```diff
@@ -1,81 +1,81 @@
...
-              </div>
-              <div class="col-xl"
-                :key="field.propertyName"
-                :class="
-                  expressCell(field.expression)
-                ">
+                v-model="field.value"
+                :field="field" />
+              <template v-else>
+                <div class="col"
```
Comment (jeanpaulkommers) [inline src/components/form/FormComponent.vue:17]: Codacy has a fix for the issue: [Expected a linebreak before this attribute.](https://app.codacy.com/gh/Supershift/Sushi.Mediakiwi.UI/pullRequest?prid=8560396)
```suggestion
              <div
```

File: src/components/form/FormComponent.vue [hunk 7]
```diff
@@ -1,81 +1,81 @@
...
+                    "
+                    :for="field.propertyName"
+                    :title="field.title"
+                    :class="{
+                      mandatory: field.mandatory,
+                    }">
+                    {{ field.title }}
+                  </label>
+                </div>
+                <div class="col-xl"
```
Comment (jeanpaulkommers) [inline src/components/form/FormComponent.vue:37]: Codacy has a fix for the issue: [Expected a linebreak before this attribute.](https://app.codacy.com/gh/Supershift/Sushi.Mediakiwi.UI/pullRequest?prid=8560396)
```suggestion
              <div
```

File: src/components/form/FormComponent.vue [hunk 8]
```diff
@@ -1,84 +1,100 @@
...
-                  expressCell(field.expression)
-                ">
+                v-model="field.value"
+                :field="field"
+                @value-changed="handleFieldsChanged(field)"/>
+              <template v-else>
 
-                  <component
-                    :is="checkVueType(field)"
+                <div class="col"
```
Comment (jeanpaulkommers) [inline src/components/form/FormComponent.vue:30]: Codacy has a fix for the issue: [Expected a linebreak before this attribute.](https://app.codacy.com/gh/Supershift/Sushi.Mediakiwi.UI/pullRequest?prid=8560396)
```suggestion
              <div
```

File: src/components/form/FormComponent.vue [hunk 9]
```diff
@@ -1,84 +1,100 @@
...
+                    :for="field.propertyName"
+                    :title="field.title"
+                    :class="{
+                      mandatory: field.mandatory,
+                    }">
+                    {{ field.title }}
+                  </label>
+                </div>
+
+                <div class="col-xl"
```
Comment (jeanpaulkommers) [inline src/components/form/FormComponent.vue:51]: Codacy has a fix for the issue: [Expected a linebreak before this attribute.](https://app.codacy.com/gh/Supershift/Sushi.Mediakiwi.UI/pullRequest?prid=8560396)
```suggestion
              <div
```

File: src/utils/api-service.ts [hunk 10]
```diff
@@ -0,0 +1,213 @@
...
+  },
+  postContentMediakiwiAPI(request: PostContentMediakiwiRequestModel, url: string): Promise<PostContentMediakiwiResponseModel> {
+    const config = {
+      withCredentials: true,
+      headers: { 
+        "original-url": url,
+      }
+    };
+    return new Promise((resolve, reject) => {
+      axiosInstance.post<any>("content/PostContent", request, config)
```
Comment (jeanpaulkommers) [inline src/utils/api-service.ts:202]: Codacy has a fix for the issue: [Unexpected any. Specify a different type.](https://app.codacy.com/gh/Supershift/Sushi.Mediakiwi.UI/pullRequest?prid=8560396)
```suggestion
      axiosInstance.post<unknown>("content/PostContent", request, config)
```

Comment (jeanpaulkommers) [inline src/utils/api-service.ts:202]: Codacy has a fix for the issue: [Unexpected any. Specify a different type.](https://app.codacy.com/gh/Supershift/Sushi.Mediakiwi.UI/pullRequest?prid=8560396)
```suggestion
      axiosInstance.post<unknown>("content/PostContent", request, config)
```

Comment (jeanpaulkommers) [inline src/utils/api-service.ts:202]: Codacy has a fix for the issue: [Unexpected any. Specify a different type.](https://app.codacy.com/gh/Supershift/Sushi.Mediakiwi.UI/pullRequest?prid=8560396)
```suggestion
      axiosInstance.post<unknown>("content/PostContent", request, config)
```

Comment (jeanpaulkommers) [inline src/utils/api-service.ts:202]: Codacy has a fix for the issue: [Unexpected any. Specify a different type.](https://app.codacy.com/gh/Supershift/Sushi.Mediakiwi.UI/pullRequest?prid=8560396)
```suggestion
      axiosInstance.post<unknown>("content/PostContent", request, config)
```

Comment (jeanpaulkommers) [inline src/utils/api-service.ts:202]: Codacy has a fix for the issue: [Unexpected any. Specify a different type.](https://app.codacy.com/gh/Supershift/Sushi.Mediakiwi.UI/pullRequest?prid=8560396)
```suggestion
      axiosInstance.post<unknown>("content/PostContent", request, config)
```

Comment (jeanpaulkommers) [inline src/utils/api-service.ts:202]: Codacy has a fix for the issue: [Unexpected any. Specify a different type.](https://app.codacy.com/gh/Supershift/Sushi.Mediakiwi.UI/pullRequest?prid=8560396)
```suggestion
      axiosInstance.post<unknown>("content/PostContent", request, config)
```

Comment (jeanpaulkommers) [inline src/utils/api-service.ts:202]: Codacy has a fix for the issue: [Unexpected any. Specify a different type.](https://app.codacy.com/gh/Supershift/Sushi.Mediakiwi.UI/pullRequest?prid=8560396)
```suggestion
      axiosInstance.post<unknown>("content/PostContent", request, config)
```

Comment (jeanpaulkommers) [inline src/utils/api-service.ts:202]: Codacy has a fix for the issue: [Unexpected any. Specify a different type.](https://app.codacy.com/gh/Supershift/Sushi.Mediakiwi.UI/pullRequest?prid=8560396)
```suggestion
      axiosInstance.post<unknown>("content/PostContent", request, config)
```

**Review by jeanpaulkommers — CHANGES_REQUESTED** [review]

**Review by jonvanputten-sf — APPROVED** [review]

**Pull request merged**